src/middleware/log_request.rs

```diff
@@ -60,6 +64,104 @@ pub fn add_custom_metadata<V: Display>(req: &mut dyn RequestExt, key: &'static s
     }
 }
 
+fn report_to_sentry(req: &dyn RequestExt, res: &AfterResult, response_time: u64) {
+    let (message, level) = match res {
+        Err(e) => (e.to_string(), Level::Error),
+        Ok(_) => {
+            if response_time <= SLOW_REQUEST_THRESHOLD_MS {
+                return;
+            }
+
+            let message = format!("Slow Request: {} {}", req.method(), req.path());
+            (message, Level::Info)
+        }
+    };
+
+    let config = |scope: &mut sentry::Scope| {
+        let method = Some(req.method().as_str().to_owned());
+
+        let scheme = match req.scheme() {
+            Scheme::Http => "http",
+            Scheme::Https => "https",
+        };
+
+        let host = match req.host() {
+            Host::Name(name) => name.to_owned(),
+            Host::Socket(addr) => addr.to_string(),
+        };
+
+        let path = &req.extensions().find::<OriginalPath>().unwrap().0;
+
+        let url = format!("{}://{}{}", scheme, host, path).parse().ok();
+
+        {
+            let id = req
+                .extensions()
+                .find::<TrustedUserId>()
+                .map(|x| x.0.to_string());
+
+            let user = sentry::User {
+                id,
+                ..Default::default()
+            };
+
+            scope.set_user(Some(user));
+        }
+
+        {
+            let filtered_headers = vec!["Authorization", "Cookie", "X-Real-Ip"];
```

Comment:
Let's move this to a constant at the top of the file, like `SENTRY_REMOVED_HEADERS: &[&str]`.

Reply:
done